### PR TITLE
Enhance Arabic text normalization and update mock data for QPC Hafs script

### PR DESCRIPTION
**Pull request by YasmineEl-shahat**
## Summary

- Switch Quran text data source to **QPC Hafs** script from [Tarteel.ai resources](https://qul.tarteel.ai/resources/quran-script/86), replacing the previous `quran.json` with ayahs loaded via `QuranDataService`
- Enhance Arabic text normalization (`normalizeArabic`) to handle QPC-specific characters: alef wasla (U+0671), QPC sukun, small high forms, sajdah marks, and other extended signs
- Add RTL styling for suggestion items using the `qpc-hafs` font family
- Add unit tests covering QPC text normalization edge cases

## Motivation

Resolves https://github.com/AmmarCodes/obsidian-quran-helper-plugin/issues/19

The QPC Hafs script provides a more accurate Quran text with standardized verse keys (e.g., `1:1`), which aligns with the Uthmani script used in printed Mushafs and enables future features like verse-key-based lookups (#18).

## Changes

| File | What changed |
|---|---|
| `src/FzfSurahModal.ts` | Use `quranDataService.getAyahs()` instead of importing `quran.json` directly |
| `src/utils.ts` | Expanded normalization regex to strip QPC marks (U+06D6-U+06ED), alef wasla, and extended diacritics |
| `src/initialAyahs.ts` | Updated mock ayah text to QPC Hafs script |
| `src/ayahs.json` | Updated full ayah dataset to QPC Hafs script |
| `src/quran.json` | Removed (no longer needed) |
| `styles.css` | Added RTL styling for `.suggestion-item` with `qpc-hafs` font |
| `tests/utils.test.ts` | Added test cases for QPC-specific normalization |
| `tests/QuranSearch.test.ts` | Updated test fixtures to match new QPC text |

## Test plan

- [ ] Verify ayah search returns correct results with Arabic input
- [ ] Verify full surah insertion works with the new data source
- [ ] Confirm suggestion dropdown renders correctly with RTL/QPC font
- [ ] Run `npm test` -- all unit tests pass

### CODE REVIEW

File: src/utils.ts
Comment (AmmarCodes) [inline src/utils.ts]: Can we make sure we didn't remove previous rules and only add to it? so the normalization stays general not specific to current source.

File: styles.css [hunk 1]
```diff
@@ -17,6 +17,10 @@
   }
 }
 
+.suggestion-item [dir="rtl"] {
```
Comment (AmmarCodes) [inline styles.css]: I think this will get applied to all other `suggestion-item`, maybe we can drop this change for now and get back to it later.

File: src/utils.ts
Comment (YasmineEl-shahat) [reply src/utils.ts]: Now the diff is cleaner:
Tashkeel	\u06d6\u06d7 kept, range expanded from \u0652 → \u065E
Alef variants	Added \u0671 (alef wasla)
Small kasra → QPC marks	Replaced redundant \u061A rule (already in \u0617-\u061A range) with broader QPC marks \u06D6-\u06DC\u06DE-\u06ED

File: src/utils.ts [hunk 2]
```diff
@@ -1,21 +1,21 @@
-// Function to normalize Arabic text
+// Function to normalize Arabic text for search
 export function normalizeArabic(text: string): string {
   return (
     text
       // Remove diacritics (tashkeel)
-      .replace(/[\u0617-\u061A\u06d6\u06d7\u064B-\u0652]/g, "")
+      .replace(/[\u0617-\u061A\u06D6\u06D7\u064B-\u065E]/g, "")
```
Comment (AmmarCodes) [inline src/utils.ts]: We're missing the `\u0652` from the previous version.


```suggestion
      .replace(/[\u0617-\u061A\u06D6\u06D7\u064B-\u065E\u0652]/g, "")
```
